CONTRIBUTING.md

```diff
@@ -1,13 +1,249 @@
+   ```
+
+4. Build the project:
+   ```console
+   cargo build
+   ```
+
+5. Run the tests to ensure everything is working:
+   ```console
+   cargo test
+   ```
+
+## Development Workflow
+
+### Before Making Changes
+
+1. Sync your fork with upstream:
+   ```console
+   git fetch upstream
+   git checkout main
+   git merge upstream/main
+   ```
+
+2. Create a new branch for your changes:
+   ```console
+   git checkout -b feature/your-feature-name
+   ```
+
+### Making Changes
+
+1. Make your changes in your feature branch
+
+2. Ensure your code follows the project's coding standards (see [Coding Guidelines](#coding-guidelines))
+
+3. Write or update tests as needed
+
+4. Run the pre-commit checks before committing:
+   ```console
+   ./ci/check.sh
+   ```
+
+   This script runs:
+   - `cargo fmt` - Code formatting
+   - `cargo clippy` - Linting
+   - `cargo test` - Unit tests
+
+### Committing Changes
+
+- Write clear, concise commit messages
+- Use the imperative mood in commit messages (e.g., "Add feature" not "Added feature")
+- Reference relevant issues in your commit messages when applicable (e.g., "Fix #123")
+
+## Issue Tracker
+
+### Reporting Bugs
+
+Before submitting a bug report:
+
+1. Search existing issues to see if the problem has already been reported
+2. Try to reproduce the issue with the latest version
+
+When submitting a bug report, please include:
+
+- A clear, descriptive title
+- Steps to reproduce the issue
+- Expected behavior vs. actual behavior
+- Your environment (OS, Rust version, fqtk version)
+- Any relevant error messages or logs
+- Sample input files if applicable (or a description of the input format)
+
+### Suggesting Features
+
+Feature requests are welcome! When suggesting a feature:
+
+1. Check existing issues to avoid duplicates
+2. Clearly describe the use case and why the feature would be valuable
+3. If possible, outline how you envision the feature working
+
+### Issue Labels
+
+- `bug` - Something isn't working
+- `enhancement` - New feature or request
+- `documentation` - Documentation improvements
+- `good first issue` - Good for newcomers
+
+## Pull Requests
+
+### Before Submitting
+
+1. Ensure all tests pass locally: `cargo test`
+2. Run the full pre-commit checks: `./ci/check.sh`
+3. Update documentation if needed
+4. Add tests for new functionality
+
+### Submitting a Pull Request
+
+1. Push your branch to your fork:
+   ```console
+   git push origin feature/your-feature-name
+   ```
+
+2. Open a pull request against the `main` branch of the upstream repository
+
+3. In your PR description:
+   - Describe what changes you made and why
+   - Reference any related issues (e.g., "Closes #123")
+   - Note any breaking changes
+
+### Review Process
+
+- All PRs require review before merging
+- CI checks must pass (formatting, linting, tests)
+- Address reviewer feedback promptly
+- Keep PRs focused and reasonably sized when possible
```

Comment:
Contributors are probably more likely to read and follow these instructions if we primarily have them in a pull request template:

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository